pkg/report/crash/types.go

```diff
@@ -29,3 +36,61 @@ func (t Type) String() string {
 	}
 	return string(t)
 }
+
+type TypeGroupPred func(Type) bool
+
+func (t Type) IsKASAN() bool {
+	return t == KASAN
+}
+
+func (t Type) IsKMSAN() bool {
+	return t == KMSAN
+}
+
+func (t Type) IsKCSAN() bool {
+	return t == KCSANDataRace || t == KCSAN
+}
+
+func (t Type) IsUBSAN() bool {
+	return t == UBSAN || t == MemorySafetyUBSAN
+}
+
+func (t Type) IsBUG() bool {
+	return t == Bug || t == MemorySafetyBUG
+}
+
+func (t Type) IsWarning() bool {
+	return t == Warning || t == MemorySafetyWARNING
+}
+
+func (t Type) IsBugOrWarning() bool {
+	return t.IsBUG() || t.IsWarning()
+}
+
+func (t Type) IsMemSafety() bool {
```

Comment:
All KASAN and KMSAN bugs would also be memory-safety bugs.

I don't quite understand this classification.

Reply:
It is a part of the bigger effort. In this step I'd like to move graph.go towards crash.Type usage. There is no change in graphs.go logic.

In the next steps we'll review these categories in order to sort the bugs by impact.